ebos/eclbasevanguard.hh

```diff
@@ -801,7 +803,7 @@ protected:
     /*! \brief Mapping between cartesian and compressed cells.
      *  It is initialized the first time it is called
      */
-    std::vector<int> cartesianToCompressed_;
+    std::map<int,int> cartesianToCompressed_;
 
```

Comment:
why do you need to be a map?

Reply:
To avoid a full cart size vector.. ie only store ids for local cells.

Reply:
Somebody should check whether using a map somewhere really saves anything. Let us do some back of the envelope calculation:

For each map entry you need to store key and value (for int that should be 8 bytes) + plus the overhead of red-black tree node (for gcc that is another 3 pointers (parent, left, right) + an enum, hence more than 12 bytes per entry).

A vector only needs 4 bytes per entry. Unless we usually have  less than 20% of active cells, std::map seems an odd choice even if performance does not matter at all. We probably can assume that if we have more than 4 processes computing, but below this threshold we are very generous with our memory.

IMHO at some time this needs to done properly using a sorted vector of std::pairs or an unordered_map. Personally, I would prefer the former (but I might be a bit lonely with this opinion), because once the map is set up no further entries will be added.

Reply:
it's not just about active cells, it's about cells on this process vs global cartesian cells. as in, the vector is of full cartesian size (of the underlying grid). but anyways, i'll just drop this commit.